libs/platform/src/lib/components/form/textarea/textarea.component.ts

```diff
@@ -0,0 +1,354 @@
+
+let uniqueId = 0;
+
+/**
+ * Textarea field implementation to be compliant with our FormGroup/FormField design
+ *
+ */
+@Component({
+    selector: 'fdp-textarea',
+    templateUrl: 'textarea.component.html',
+    styleUrls: ['textarea.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [
+        {
+            provide: FormFieldControl,
+            useExisting: TextareaComponent,
+            multi: true
+        }
+    ]
+})
+export class TextareaComponent extends BaseInput {
+    @Input()
+    height: string;
+
+    @Input()
+    growingMaxLines: number; // rows
+
+    @Input()
+    cols: number;
+
+    @Input()
+    wrap: WrapType = 'soft';
+
+    @Input()
+    readonly: boolean;
+
+    @Input()
+    maxLength: number;
+
+    @Input()
+    showExceededText: boolean = false;
+
+    @Input()
+    growing: boolean = false;
+
+    @Input()
+    textExceededMessage: string;
+
+    /**
+     *  The state of the form control - applies css classes.
+     *  Can be `success`, `error`, `warning`, `information` or blank for default.
+     */
+    @Input()
+    stateType: FormStates = 'warning';
+
+    @Input() get value(): any {
+        // if (super.getValue() == null || !super.getValue()) {
+        //     this.value = '';
+        // }
+        return super.getValue();
+    }
+
+    set value(value: any) {
+        if (value) {
+            super.setValue(value);
+        } else {
+            // when custom value not set, we should set/reset counter to maxlength value when it becomes undefined
+            this.exceededCharCount = this.maxLength ? this.maxLength : 0;
+            // reset state by resetting value
+            super.writeValue('');
+        }
+    }
+
+    /** @hidden */
+    @ViewChild('textareaElement')
+    textareaElement: ElementRef;
+
+    /** @hidden */
+    hasTextExceeded: boolean = false;
+
+    /** excess character count */
+    exceededCharCount: number = this.maxLength ? this.maxLength : 0;
+
+    counterExcessOrRemaining = 'remaining';
+
+    isValueCustomSet = false;
+
+    /** @hidden */
+    /** to track the current textarea content */
+    // textAreaValue: string;
+
+    /** @hidden */
+    /** to keep track of number of characters in the textarea */
+    private _textAreaCharCount: number = 0;
+
+    /** @hidden */
+    private _isPasted: boolean = false;
+
+    /**for auto grow calculation */
+
+    private _maxHeightStr: number;
+
+    /** for i18n counter message translation */
+
+    private readonly remainingText = 'remaining';
+    private readonly excessText = 'excess';
+
+    constructor(
+        protected _cd: ChangeDetectorRef,
+        @Optional() @Self() public ngControl: NgControl,
+        @Optional() @Self() public ngForm: NgForm
+    ) {
+        super(_cd, ngControl, ngForm);
+        if (this.ngControl) {
+            this.ngControl.valueAccessor = this;
+        }
+        this.id = `textarea-${uniqueId++}`;
```

Comment:
remove it from contructor - just intialize just like other inputs.

Reply:
removed